### DISCUSSION TITLE
Table-Manager wie verlinkt man vom Frontend in den Datensatz?

### DISCUSSION THREAD

**Discussion by skerbis** [FriendsOfREDAXO/.github#15] [Q&A] [answered]
*Szenario:*
Ich versuche aus dem Frontend in die Editierung eines Datensatzes zu gelangen. Leider wird der CSRF-Token nicht akzeptiert.

*code*

```php
if (rex_backend_login::hasSession() && $newsDataId != "") // REDAXO 5 Backend - Login - Check
{
  $table = rex_yform_manager_table::get('rex_news');
  $_csrf_key = $table->getCSRFKey();
  $_csrf_params = rex_csrf_token::factory($_csrf_key)->getUrlParams();
  $token =  $_csrf_params['_csrf_token'];    

    $edit = ' <a style="" class="uk-button uk-button-primary" href="' . rex_url::backendPage(
        'yform/manager/data_edit',
        [
            'table_name' => 'rex_news',
            'func' => 'edit',
            'data_id' => $newsDataId,
            '_csrf_token' => $token
            
        ]
    ) . '">
<i class="fal fa-edit"></i>
News bearbeiten</a>
                ';
}
```

**Comment by skerbis** [comment, marked as answer]
*Lösung*

Die CSRF-Tokens für das Frontend und für das Backend unterscheiden sich. 
Hier hilft es die Umgebung temporär umzuschalten so dass die CSRF-Generierung in der richtigen Umgebung ausgeführt wird. 


Vor Erstellung des CSRF-Codes 

```php
rex::setProperty('redaxo', true);
```

ausführen.

Danach wieder mit 
```php
rex::setProperty('redaxo', false);
```
zurücksetzen

**Comment by AWqxKAWERbXo** [comment]
Achtung, es muss auch ggf. geprüft werden, ob man in der CLI-Umgebung ist - dort gibt es das `$_REQUEST`-Objekt nicht. <https://github.com/redaxo/redaxo/issues/6127#issuecomment-2241770195>

**Comment by skerbis** [comment]
Als Ergänzung es geht noch  etwas einfacher. https://github.com/orgs/FriendsOfREDAXO/discussions/42 , die Properties müssen dennoch gesetzt werden.
